### ISSUE TITLE
OADP-1668: Volumesnapshot related CR’s namely Volumesnapshot and VolumeSnapshotcontent are not being included by the OADP Version of Velero server in the backup bundle

### ISSUE THREAD

**Issue by vjaincatalogic** [openshift/oadp-operator#974]
### Only developers will use GitHub issues for development purposes

agree

**Comment by vjaincatalogic** [comment]
I attach the backup bundle here:
[test-backup-8-csi-volumesnapshotclasses.json.gz](https://github.com/openshift/oadp-operator/files/11232181/test-backup-8-csi-volumesnapshotclasses.json.gz)
[test-backup-8-csi-volumesnapshotcontents.json.gz](https://github.com/openshift/oadp-operator/files/11232183/test-backup-8-csi-volumesnapshotcontents.json.gz)
[test-backup-8-csi-volumesnapshots.json.gz](https://github.com/openshift/oadp-operator/files/11232184/test-backup-8-csi-volumesnapshots.json.gz)
[test-backup-8-logs.gz](https://github.com/openshift/oadp-operator/files/11232185/test-backup-8-logs.gz)
[test-backup-8-podvolumebackups.json.gz](https://github.com/openshift/oadp-operator/files/11232186/test-backup-8-podvolumebackups.json.gz)
[test-backup-8-resource-list.json.gz](https://github.com/openshift/oadp-operator/files/11232187/test-backup-8-resource-list.json.gz)
[test-backup-8-volumesnapshots.json.gz](https://github.com/openshift/oadp-operator/files/11232188/test-backup-8-volumesnapshots.json.gz)

**Comment by kaovilai** [comment]
Thanks for your report. tracking in [JIRA](https://issues.redhat.com/browse/OADP-1668)

**Comment by kaovilai** [comment]
@vjaincatalogic is this from building from source? or are you using official releases?

**Comment by vjaincatalogic** [comment]
@kaovilai , Yes this is from official release. I'm not building from source. I used the official oadp-operator from market place.

**Comment by kaovilai** [comment]
Can you verify version?

**Comment by vjaincatalogic** [comment]
oadp version is v1.1.3

**Comment by vjaincatalogic** [comment]
Hi @shubham-pampattiwar, I will provide you all these details but currently the cluster was deleted and it will take me some time to get back to you.

**Comment by vjaincatalogic** [comment]
backup Bundle:
[backup_bundle.tar.gz](https://github.com/openshift/oadp-operator/files/11308282/backup_bundle.tar.gz)

I provide you the first three, backup CR used, DPA CR used + DPA CR status, velero logs 
[requested.tar.gz](https://github.com/openshift/oadp-operator/files/11308431/requested.tar.gz)

volume-snapshot-mover logs and VolumeSnapshotBackup CR and its status , I'm not able to find as I had not used Volume Snapshot Mover, IS it the reason, I'm not able to see the volumesnapshot in backup bundle in my BSL?

**Comment by shubham-pampattiwar** [comment]
@vjaincatalogic Which type of OADP backup are you trying to take here ? Native CSI backup or Datamover CSI backup ?

**Comment by draghuram** [comment]
@shubham-pampattiwar, this was CSI backup. By "Datamover" CSI backup, do you mean Restic or Kopia backup?

**Comment by kaovilai** [comment]
>  "Datamover" CSI backup, do you mean Restic or Kopia backup?

We means our own implementation of "Data Mover" in OADP v1.1 which takes CSI snapshots, and make a restic copy of data from snapshot to S3. This is an OADP specific capability which upstream will have with https://github.com/vmware-tanzu/velero/pull/5968

See doc at
https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html-single/backup_and_restore/index#oadp-using-data-mover-for-csi-snapshots_backing-up-applications

TL;DR: we do not mean restic or kopia.

**Comment by kaovilai** [comment]
If you do not have `enable: true` here then you did not use our Data Mover.
```
apiVersion: oadp.openshift.io/v1alpha1
kind: DataProtectionApplication
metadata:
  name: velero-sample
  namespace: openshift-adp
spec:
  features:
    dataMover:
      enable: true
```

**Comment by kaovilai** [comment]
Should not be required.

**Comment by vjaincatalogic** [comment]
In that case, I'm just confirming from the OADP,
OADP team got all the information required to investigate on this issue and OADP team is still investigating?

**Comment by kaovilai** [comment]
At this time we are not requesting any more information. This issue is still open to investigation.

**Comment by vjaincatalogic** [comment]
Hi Team, Any Updates on this issue?

**Comment by weshayutin** [comment]
We had to push it out further. Tracked in OADP-1669

**Comment by vjaincatalogic** [comment]
Hi, I retried the complete flow with OADP version - 1.1.4 as well but still the same issue, I see that volumesnapshot is null, 
[test-backup-11-csi-volumesnapshotclasses.json.gz](https://github.com/openshift/oadp-operator/files/11787511/test-backup-11-csi-volumesnapshotclasses.json.gz)
[test-backup-11-csi-volumesnapshotcontents.json.gz](https://github.com/openshift/oadp-operator/files/11787512/test-backup-11-csi-volumesnapshotcontents.json.gz)
[test-backup-11-csi-volumesnapshots.json.gz](https://github.com/openshift/oadp-operator/files/11787513/test-backup-11-csi-volumesnapshots.json.gz)
[test-backup-11-logs.gz](https://github.com/openshift/oadp-operator/files/11787514/test-backup-11-logs.gz)
[test-backup-11-podvolumebackups.json.gz](https://github.com/openshift/oadp-operator/files/11787515/test-backup-11-podvolumebackups.json.gz)
[test-backup-11-resource-list.json.gz](https://github.com/openshift/oadp-operator/files/11787516/test-backup-11-resource-list.json.gz)
[test-backup-11-volumesnapshots.json.gz](https://github.com/openshift/oadp-operator/files/11787517/test-backup-11-volumesnapshots.json.gz)
[test-backup-11.tar.gz](https://github.com/openshift/oadp-operator/files/11787518/test-backup-11.tar.gz)
[velero-backup.json.gz](https://github.com/openshift/oadp-operator/files/11787519/velero-backup.json.gz)

Dataprotectionapp manifest: 
[dataprotection.yaml.gz](https://github.com/openshift/oadp-operator/files/11787634/dataprotection.yaml.gz)
[velero.log.gz](https://github.com/openshift/oadp-operator/files/11787635/velero.log.gz)
[backup.yaml.gz](https://github.com/openshift/oadp-operator/files/11787637/backup.yaml.gz)

**Comment by kaovilai** [comment]
Still looking into it. Thanks for the details!
